### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by malaikaaiyar**
I added more instructions on how to get the server running on someone's own computer. I hope I have done this right.

### CODE REVIEW

File: README.md
Comment (0xdevalias) [inline README.md:16]: No need to comment out these bits, can just delete them.

(only commenting on the first, but this applies to all of the same pattern of issue)

File: README.md
Comment (0xdevalias) [inline README.md:18]: Since this repo is also open-source, we want to make sure that the instructions are also useful to people outside of Sparkle as well.

Keeping the note about saving it as `.env.development.local` (the previous suggestion of `.env.staging` is wrong) is a good idea.

Potentially we could also link to https://create-react-app.dev/docs/adding-custom-environment-variables#adding-development-environment-variables-in-env to help explain what `.env` files can be used/why/etc.

An empty `.env.example` file exists at https://github.com/sparkletown/sparkle/blob/staging/.env.example that shows the format it needs to be in. So perhaps saying something like about copying `.env.example` to `.env.development.local`, and then filling it in with the appropriate details. For Sparkle, they can get that off other engineers. For random public, the main required variables relate to the deployed firebase project, so they can get them from there (the what/how of deploying this for the first time is probably out of scope (OOS) of this section of the README, so don't worry going into detailed specifics there)

File: README.md
Comment (0xdevalias) [inline README.md:29]: They are equivalent, but this is more 'obvious' what it's doing, so will help new people more

```suggestion
npm install
```

File: README.md
Comment (0xdevalias) [inline README.md:9]:
```suggestion
First, clone the repo and `cd` into it:
```

File: README.md
Comment (0xdevalias) [reply README.md:18]: Since we will pretty much always use the same environment for local dev, we probably don't need to mention "for the environment" here either.

File: README.md
Comment (0xdevalias) [inline README.md:26]:
```suggestion
Install the platform dependencies with `npm`:

(**Note:** `npm` v7+ is not supported, it will cause issues with our `package-lock.json`, and you may end up with the wrong dependency versions)
```

File: README.md
Comment (0xdevalias) [inline README.md:35]: We don't need `npm test` here. Technically we don't really have many/any tests at the moment anyways. But it might still be worth mentioning this command somewhere later on, perhaps near the "you can also test builds" section.

```suggestion
```

File: README.md
Comment (0xdevalias) [inline README.md:39]:
```suggestion
While you generally won't need to do this while developing locally, you can manually build the platform assets as follows:
```

File: README.md
Comment (0xdevalias) [inline README.md:48]:
```suggestion
**Note:** Before you run the following steps, you will need to ensure you have access to the Firebase project you want to use. This access can be set up through the Firebase web UI.

In a new terminal, from the directory you cloned the code to, enter the following commands:
```

File: README.md
Comment (0xdevalias) [inline README.md:51]: Technically this doesn't need to be installed globally, as we include it as part of our dev dependencies, so it will be installed/usable in that way regardless.

```suggestion
# While not necessary (as we already include it in our devDependencies), you can install the firebase-tools globally if desired
# npm install -g firebase-tools@latest

```

File: README.md
Comment (0xdevalias) [inline README.md:56]:
```suggestion
# Move into the firebase functions directory
cd functions

# Install the function dependencies
npm install

# Login to firebase with the account that has access to this project
firebase login

# Switch to the 'staging' project in our local configuration (or whichever environment you are developing against)
firebase use staging

# Copy the runtime config locally
npm run firebase functions:config:get > .runtimeconfig.json
```

File: README.md
Comment (0xdevalias) [inline README.md:60]: Merged into above suggestion.

```suggestion
```

File: README.md
Comment (0xdevalias) [inline README.md:65]:
```suggestion
npm run firebase:emulate-functions

# Or if you don't want to use our helper scripts, you can do this directly:
# firebase emulators:start --only functions
```

File: README.md
Comment (0xdevalias) [inline README.md:61]:
```suggestion
Now you're ready to launch the backend function emulator! ✨
```

File: README.md
Comment (0xdevalias) [inline README.md:36]: Technically I think that we need to be emulating the firebase functions locally (as below) before the server can properly start/the code work properly. Not 100% about this. It may only be required when we don't want to use the _actual_ staging functions (eg. when developing changes to the functions)

Not sure of the best way to fit that knowledge into this README (or if it's even the right place to). Let's see what you come up with :)

**Review by 0xdevalias — COMMENTED** [review]
Thanks for opening a PR to update this documentation!

Added a bunch of thoughts/suggestions (don't take the number of comments/etc personally, I tend to just be quite detail hyperfocused in my reviews, which i'm sure the other devs could tell you all about :p)

File: README.md
Comment (malaikaaiyar) [reply README.md:37]: So would someone need to do the firebase steps before the frontend ones?

Comment (malaikaaiyar) [reply README.md:37]: I did the steps in the order that they came in originally and everything seems to work fine (I think)

**Comment by 0xdevalias** [comment]
@malaikaaiyar Thanks for pushing those updates, I'll re-review today.

Also, for future note, when you've made changes and it's ready to be re-reviewed, you can click this button to make sure we get notified/etc:

- image

Further reading:

- https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review

File: README.md
Comment (0xdevalias) [reply README.md:31]: > So would someone need to do the firebase steps before the frontend ones?

That was sort of what I was pondering/wondering when I asked my above question.

> I did the steps in the order that they came in originally and everything seems to work fine (I think)

Cool, if it worked that way, then probably should be fine as is :)

File: README.md
Comment (0xdevalias) [inline README.md:16]:
```suggestion
In the main `sparkle` folder, copy `.env.example` to a new file `.env.local`, and fill it with the appropriate details.
```

File: README.md
Comment (0xdevalias) [inline README.md:18]:
```suggestion
You can read more about the various `.env` files that you can use at:

- https://create-react-app.dev/docs/adding-custom-environment-variables#adding-development-environment-variables-in-env
```

File: README.md
Comment (0xdevalias) [inline README.md:74]:
```suggestion

**Note**: You might need to emulate the firebase functions locally before the server can properly start. If you have issues using/editing the actual staging functions, try that.
```

**Review by 0xdevalias — APPROVED** [review]
A couple of tiny changes, then this will be good to merge! Congrats on your first PR/contribution! :)

@malaikaaiyar I'm preemptively approving this so that you don't have to wait for another round of timezone-delayed reviews, but it would be good if you can make the suggested changes before you do so.

Feel free to ask one of the other @sparkletown/engineering people to give it a final look over if you're not 100% sure.

Once you're ready to merge it, go down to the merge button, and choose 'squash merge' (we always squash merge into the `staging` branch, and we always 'normal merge' when 'deploying' code to the `master` branch). Generally I will keep the default title/description it suggests there.

**Pull request merged**
